File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-4.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    findMin(nums) {
        let left = 0;
        let right = nums.length - 1;

        while (left < right) {
            const middle = left + Math.floor((right - left) / 2);
            if (nums[middle] < nums[right]){
                right = middle;
            } else {
                left = middle + 1;
            }
        }
        return nums[left];
    }
}

File: Data Structures & Algorithms/binary-search/submission-4.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        left = 0
        right = len(nums)

        while left < right:
            middle = left + (right - left) // 2
            current = nums[middle]

            if current == target:
                return middle
            elif current < target:
                left = middle + 1
            elif current > target:
                right = middle

        return -1